### PR TITLE
Additional switch logs

### PR DESCRIPTION
**Pull request by jacekchalupka**

### CODE REVIEW

**Comment by jacekchalupka** [comment]
recheck no bug

File: Test/TestConfigurationUtils.ps1 [hunk 1]
```diff
@@ -205,15 +205,46 @@ function Test-IfVmSwitchExist {
...
         Get-VMSwitch $Using:VMSwitchName -ErrorAction SilentlyContinue | Select-Object -ExpandProperty isDeleted
     }
 
-    if(($null -eq $r) -or ($r.Equals($true))) {
+    if($null -eq $r) {
+        Write-Log '... it does not.'
         return $false
     }
+    elseif($r.Equals($true)) {
+        Write-Log '... it still exist, but is deleting.'
```
Comment (michal-clapinski) [inline Test/TestConfigurationUtils.ps1]:
```suggestion
        Write-Log '... it still exists, but is being deleted.'
```

File: Test/TestConfigurationUtils.ps1 [hunk 2]
```diff
@@ -205,15 +205,46 @@ function Test-IfVmSwitchExist {
...
+        return $false
+    }
+    elseif($r.Equals($false)) {
+        Write-Log '... it still exist.'
```
Comment (michal-clapinski) [inline Test/TestConfigurationUtils.ps1]:
```suggestion
        Write-Log '... it still exists.'
```

File: Test/TestConfigurationUtils.ps1 [hunk 3]
```diff
@@ -205,15 +205,46 @@ function Test-IfVmSwitchExist {
...
+        return $true
+    }
 
-     return $true
+    Write-Log "... it returned: $r"
+    throw "Checking if switch exist failed."
```
Comment (michal-clapinski) [inline Test/TestConfigurationUtils.ps1]:
```suggestion
    throw "Checking if switch exists failed."
```

File: Test/TestConfigurationUtils.ps1 [hunk 4]
```diff
@@ -205,15 +205,46 @@ function Test-IfVmSwitchExist {
...
+}
+
+function Get-IPAddresses {
```
Comment (michal-clapinski) [inline Test/TestConfigurationUtils.ps1]:
```suggestion
function Write-IpAddresses {
```

File: Test/TestConfigurationUtils.ps1 [hunk 5]
```diff
@@ -205,15 +205,46 @@ function Test-IfVmSwitchExist {
...
-    if(($null -eq $r) -or ($r.Equals($true))) {
+    if($null -eq $r) {
+        Write-Log '... it does not.'
         return $false
     }
+    elseif($r.Equals($true)) {
+        Write-Log '... it still exist, but is deleting.'
+        return $false
+    }
+    elseif($r.Equals($false)) {
```
Comment (michal-clapinski) [inline Test/TestConfigurationUtils.ps1]: Why not just `else`?

File: Test/TestConfigurationUtils.ps1 [hunk 6]
```diff
@@ -225,6 +256,8 @@ function Assert-VmSwitchInitialized {
         throw "VmSwitch is not created. No need to wait for IP on $($SystemConfig.VHostName)."
     }
 
+    Get-IPAddresses -Session $Session -SystemConfig $SystemConfig
```
Comment (jacekchalupka) [inline Test/TestConfigurationUtils.ps1]:
```suggestion
    Write-IPAddresses -Session $Session -SystemConfig $SystemConfig
```

File: Test/TestConfigurationUtils.ps1 [hunk 7]
```diff
@@ -237,6 +270,8 @@ function Assert-VmSwitchDeleted {
         throw "VmSwitch is not going to be deleted. No need to wait for IP on $($SystemConfig.AdapterName)."
     }
 
+    Get-IPAddresses -Session $Session -SystemConfig $SystemConfig
```
Comment (jacekchalupka) [inline Test/TestConfigurationUtils.ps1]:
```suggestion
    Write-IPAddresses -Session $Session -SystemConfig $SystemConfig
```

File: Test/TestConfigurationUtils.ps1 (hunk 5 above)
Comment (jacekchalupka) [reply Test/TestConfigurationUtils.ps1]: Just to be sure that it returned false not just some mambo jumbo.

**Comment by jacekchalupka** [comment]
recheck no bug

**Comment by jacekchalupka** [comment]
recheck no bug

**Pull request merged**